Sources/SwiftDocC/Model/MarkdownOutput/Translation/MarkdownOutputSemanticVisitor.swift

```diff
@@ -0,0 +1,442 @@
+            let declaration = declarationFragments
+                .map { $0.spelling }
+                .joined()
+            let code = CodeBlock(declaration)
+            markdownWalker.visit(code)
+        }
+        
+        if let parametersSection = symbol.parametersSection, parametersSection.parameters.isEmpty == false {
+            markdownWalker.visit(Heading(level: 2, Text(ParametersSection.title ?? "Parameters")))
+            for parameter in parametersSection.parameters {
+                markdownWalker.visit(Paragraph(InlineCode(parameter.name)))
+                markdownWalker.visit(container: MarkupContainer(parameter.contents))
+            }
+        }
+        
+        markdownWalker.visit(section: symbol.returnsSection)
+        
+        markdownWalker.visit(section: symbol.discussion, addingHeading: symbol.kind.identifier.swiftSymbolCouldHaveChildren ? "Overview" : "Discussion")
+        
+        markdownWalker.outgoingReferences = []
+        markdownWalker.withRenderingLinkList {
+            $0.visit(section: symbol.topics, addingHeading: "Topics")
+            $0.visit(section: symbol.seeAlso, addingHeading: "See Also")
+        }
+        
+        manifest?.relationships.formUnion(markdownWalker.outgoingReferences)
+        
+        for relationshipGroup in symbol.relationships.groups {
+            for destination in relationshipGroup.destinations {
+                switch context.resolve(destination, in: identifier) {
+                case .success(let resolved):
+                    add(target: resolved, type: .relatedSymbol, subtype: relationshipGroup.kind.rawValue)
+                case .failure:
+                    if let fallback = symbol.relationships.targetFallbacks[destination] {
+                        add(fallbackTarget: fallback, type: .relatedSymbol, subtype: relationshipGroup.kind.rawValue)
+                    }
+                }
+            }
+        }
+        return MarkdownOutputNode(metadata: metadata, markdown: markdownWalker.markdown)
+    }
+}
+
+import SymbolKit
+
+extension MarkdownOutputNode.Metadata.Symbol {
+    init(_ symbol: SwiftDocC.Symbol, context: DocumentationContext, bundle: DocumentationBundle) {
+                
+        // Gather modules
+        var modules = [String]()
+
+        if let main = try? context.entity(with: symbol.moduleReference) {
+            modules.append(main.name.plainText)
+        }
+        if let crossImport = symbol.crossImportOverlayModule {
+            modules.append(contentsOf: crossImport.bystanderModules)
+        }
+        if let extended = symbol.extendedModuleVariants.firstValue, modules.contains(extended) == false {
+            modules.append(extended)
+        }
+        self.init(
+            kind: symbol.kind.identifier.identifier,
+            preciseIdentifier: symbol.externalID ?? "",
+            modules: modules
+        )
+    }
+}
+
+extension MarkdownOutputNode.Metadata.Availability {
```

Comment:
Question: Are these used outside of this file? Otherwise this could be a `private` extension, to keep this code isolated.